docs/spec/lsm/adr-002.md

```diff
@@ -0,0 +1,362 @@
+```go
+func (k Keeper) SetTotalLiquidStakedTokens(ctx sdk.Context, tokens sdk.Dec)
+func (k Keeper) GetTotalLiquidStakedTokens(ctx sdk.Context) sdk.Dec
+```
+
+### Helper functions
+In order to identify whether a liquid stake transaction will exceed either the global liquid staking cap or the validator bond cap, the following functions were added:
+
+```go
+// Check if an account is a owned by a liquid staking provider
+// This is determined by checking if the account is a 32-length module account
+func (k Keeper) AccountIsLiquidStakingProvider(address sdk.AccAddress) bool 
+
+// SafelyIncreaseTotalLiquidStakedTokens increments the total liquid staked tokens
+// if the caps are enabled and the global cap is not surpassed by this delegation
+func (k Keeper) SafelyIncreaseTotalLiquidStakedTokens(ctx sdk.Context, amount sdk.Int) error 
+
+// DecreaseTotalLiquidStakedTokens decrements the total liquid staked tokens
+// if the caps are enabled
+func (k Keeper) DecreaseTotalLiquidStakedTokens(ctx sdk.Context, amount sdk.Int) error
+
+// SafelyIncreaseValidatorTotalLiquidShares increments the total liquid shares on a validator
+// if the caps are enabled and the validator bond cap is not surpassed by this delegation
+func (k Keeper) SafelyIncreaseValidatorTotalLiquidShares(ctx sdk.Context, validator types.Validator, shares sdk.Dec) error 
+
+// DecreaseValidatorTotalLiquidShares decrements the total liquid shares on a validator
+// if the caps are enabled
+func (k Keeper) DecreaseValidatorTotalLiquidShares(ctx sdk.Context, validator types.Validator, shares sdk.Dec) error
+
+// SafelyDecreaseValidatorBond decrements the total validator's self bond
+// so long as it will not cause the current delegations to exceed the threshold
+// set by validator bond factor
+func (k Keeper) SafelyDecreaseValidatorBond(ctx sdk.Context, validator types.Validator, shares sdk.Dec) error 
+```
+
+### Accounting
+Tracking the total liquid stake and total liquid validator shares requires additional accounting changes in the following transactions/events:
+
+```go
+func Delegate() {
+    ...
+    // If delegator is a liquid staking provider
+    //    Increment total liquid staked
+    //    Increment validator liquid shares
+}
+
+func Undelegate() {
+    ...
+    // If delegator is a liquid staking provider
+    //    Decrement total liquid staked
+    //    Decrement validator liquid shares
+}
+
+func BeginRedelegate() {
+    ...
+    // If delegator is a liquid staking provider
+    //    Decrement source validator liquid shares
+    //    Increment destination validator liquid shares
+}
+
+func TokenizeShares() {
+    ...
+    // If delegator is a NOT liquid staking provider (otherwise the shares are already included)
```

Comment:
What's the reasoning behind allowing delegations from a LS provider to be tokenized? This would enable the following scenario: Let `x` and `y` be two LS providers. A user liquid stakes atoms with `A` and gets `x-atoms` in return. `x` stakes the atoms with validator `V`. Then, `x` tokenizes the staked atoms and gets the equivalent of `V/id` tokens. This tokens are liquid staked with `y` in return for `y-atoms`. Now we have both `x-atoms` and `y-atoms` for the same amount of atoms.

Reply:
This is an example of how a poorly designed liquid staking provider can become insolvent. But it's not specific to LSM, there are plenty of ways poorly designed liquid staking providers can become insolvent (e.g. failing to detect slashes). 

In your example, if designed properly, liquid staking provider `x` would need to burn `x-atoms` before _"x tokenizes the staked atoms and gets the equivalent of V/id tokens. This tokens are liquid staked with y"_, so at the end of the flow there would only be `y-atoms`. 

I would advocate for leaving the design as it's currently built, allowing liquid staking providers to tokenize shares. It's convenient for liquid staking providers to be able to tokenize their stake because it allows them to offer instant redemptions to users who desire to receive LSM shares.

Reply:
Agree with Riley - `x-atoms` would depeg (become worthless) if an LST provider did this.

As a design principal, if market forces/incentives will naturally prevent an issue, I don't think it needs to be protected against in code.

Reply:
Resolving this.